lib/pymedphys/_experimental/streamlit/apps/transfer_check.py

```diff
@@ -70,199 +61,500 @@ def main():
+    conformity_index = calc_conformity_index(dd_input, dvh_calcs, target, rx_dose)
+    homogeneity_index = calc_homogeneity_index(dvh_calcs, target, rx_dose)
+    dhi = calc_dose_homogeneity_index(dvh_calcs, target)
+
+    data = {
+        "Min [Gy]": min_dose,
+        "Max [Gy]": max_dose,
+        "Mean [Gy]": mean_dose,
+        "D1 [Gy]": D_1,
+        "D2 [Gy]": D_2,
+        "D98 [Gy]": D_98,
+        "D99 [Gy]": D_99,
+        "CI": conformity_index,
+        "HI": homogeneity_index,
+        "DHI": dhi,
+    }
+
+    target_df = pd.DataFrame.from_dict(
+        data, orient="Index", columns=[target]
+    ).style.set_precision(2)
+    return st.write(target_df)
+
+
+def compare_to_historical_scores(constraints_df, institutional_history):
+    df = constraints_df.copy()
+    df["Institutional Average"] = "-"
+    for index in df.index:
+        row_key = df["Structure_Key"][index]
+        row_type = df["Type"][index]
+        row_dose = df["Dose [Gy]"][index]
+        row_volume = df["Volume [%]"][index]
+        df["Institutional Average"][index] = float(
+            institutional_history[
+                (institutional_history["Structure_Key"] == row_key)
+                & (institutional_history["Type"] == row_type)
+                & (institutional_history["Dose [Gy]"] == row_dose)
+                & (institutional_history["Volume [%]"] == row_volume)
+            ]["Score"].mean()
+        )
+
+    return df
+
+
+def show_dicom(dicom_table):
+    show_table = st.checkbox("View complete DICOM table.")
+    if show_table:
+        st.subheader("DICOM Table")
+        st.dataframe(dicom_table, height=1000)
+
+
+def show_mosaiq(mosaiq_table):
+    show_table = st.checkbox("View complete Mosaiq table.")
+    if show_table:
+        st.subheader("Mosaiq Table")
+        st.dataframe(mosaiq_table, height=1000)
+
+
+def add_alias(dvh_calcs, ALIASES):
+    define_alias = st.checkbox("Define a new structure alias")
+    if define_alias:
+        add_new_structure_alias(dvh_calcs, ALIASES)
+
+
+def add_to_database(constraints_df, institutional_history):
+    add_to = st.checkbox("Add patient to Database")
+    if add_to:
+        add_constraint_results_to_database(constraints_df, institutional_history)
+
+
+def main():
+    server = "PRDMOSAIQIWVV01.utmsa.local"
+    connection = get_cached_mosaiq_connection(server)
+
+    st.sidebar.header("Instructions:")
+    st.sidebar.markdown(
+        """
+    To use this application, you must have the RP file of the plan you want to check. This can be exported in Pinnacle.
+    You will get an error if you select a QA RP file.
+
+    When exporting the DICOM, only the RP is needed for the data transfer check. Once you have that, you can select it where prompted and the application
+    will run.
+
+    To evaluate the DVH, you will need to import the RD and RS files as well.
+    """
+    )
+
+    files = get_patient_files()
+
+    if "rp" in files:
```

Comment:
```suggestion
    if not "rp" in files:
        st.stop()
```

Then the nesting of all the below can be reduced by 1.